pkg/services/gitops/uninstall_test.go

```diff
@@ -14,6 +16,55 @@ import (
 
 var uninstallParams gitops.UinstallParams
 
+func checkFluxUninstallFailure() {
+	fluxErrMsg := "flux uninstall failed"
+	globalErrMsg := "errors occurred during uninstall; the original state of the cluster may not be completely restored"
+
+	loggedMsg := ""
+	logger.PrintfStub = func(str string, args ...interface{}) {
+		loggedMsg = fmt.Sprintf(str, args...)
+	}
+
+	fluxClient.UninstallStub = func(namespace string, dryRun bool) error {
+		return errors.New(fluxErrMsg)
+	}
+
+	err := gitopsSrv.Uninstall(uninstallParams)
+
+	Expect(loggedMsg).To(Equal(fmt.Sprintf("received error uninstalling flux: %q, continuing with uninstall", fluxErrMsg)))
+	Expect(err.Error()).To(Equal(globalErrMsg))
+	Expect(kubeClient.GetClusterStatusCallCount()).To(Equal(1))
+	Expect(fluxClient.UninstallCallCount()).To(Equal(1))
+	namespace, dryRun := fluxClient.UninstallArgsForCall(0)
+	Expect(namespace).To(Equal("wego-system"))
+	Expect(dryRun).To(Equal(false))
+}
+
+func checkAppCRDUninstallFailure() {
+	crdErrMsg := "CRD uninstall failed"
+	globalErrMsg := "errors occurred during uninstall; the original state of the cluster may not be completely restored"
```

Comment:
Is this a string that needs to stay in sync with the other error message for this tests to pass? A custom error type might help here.

Reply:
Yep. Good call.